### PR TITLE
test(gptme-sessions): retain Pi edge-case fixtures

### PR DESCRIPTION
**Pull request by TimeToBuildBob**
## Summary

- retain sanitized real Pi v3 fixtures for failed tools, provider-error recovery, aborts, compaction, and branch topology
- assert truthful error/stop/usage and active-branch normalization against those fixtures
- add an explicit Pi 0.84.4 parser/model-catalog compatibility sentinel that fails closed on schema, stop-reason, route-catalog, or fixture-model drift

## Scope

This closes the retained fixture matrix and parser/model-catalog drift gate. Pi `file://` `sessionDir` parity remains deliberately separate and nonblocking. Automatic Pi routing remains disabled.

## Verification

- `uv run python3 -m pytest packages/gptme-sessions/tests -q --disable-warnings --maxfail=1` — 1,305 passed
- `uv run python3 packages/gptme-sessions/scripts/check_pi_compat.py` — pinned v0.84.4 contract/catalog check passed
- Ruff and MyPy passed for changed Python files
- exact-file prek hooks passed

### CODE REVIEW

File: packages/gptme-sessions/scripts/check_pi_compat.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+)
+PINNED_STOP_REASONS = frozenset(
+    {"pending", "stop", "length", "toolUse", "error", "aborted", "deferred"}
+)
+PINNED_MODEL_CATALOG_HASHES = {
+    ".manifest.json": "68eafe33e43efce9dac778d9290ed75ff65ed680d45828a1aa26b947b2fef237",
+    "openai-codex.json": "2712c2924a4a75213dddc743c0e5f08d50a781fe807f16d5afb5fb65b41c64c7",
+    "xai.json": "9668b607ac69237089e84efa3f0590dc1e28d8e24c9d1afa3a2b7cf72efa30e0",
+}
+PINNED_FIXTURE_MODELS = frozenset({("openai-codex", "gpt-5.6-luna"), ("xai", "grok-4.6")})
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply packages/gptme-sessions/scripts/check_pi_compat.py]: Addressed in 3b63e1b0.

The sentinel now pins the live models from the retained fixtures (`claude-opus-4-5`, `claude-opus-4-6`, `gpt-5.4`, plus the original Codex/xAI routes) and hashes `anthropic.json`. Two fixtures use models already gone from Pi 0.84.4 (`google-antigravity`/`claude-opus-4-5-thinking`, `openai-codex`/`gpt-5.3-codex`); those stay parser-shape coverage and are explicitly retired from the fail-closed pin.

A completeness test now fails if a retained fixture model is neither pinned nor retired, or if a live fixture provider has no hashed catalog.

**Comment by TimeToBuildBob** [comment]
@greptileai review

**Comment by TimeToBuildBob** [comment]
Merged as `b010b6c7`. Fixture-model pin is in; Greptile 5/5.

Parser closeout: remaining `file://` `sessionDir` parity is split to `pi-sessiondir-file-uri-parity` and is not a routing gate. No further review/CI action on this PR.
